packages/grid_client/src/modules/k8s.ts

```diff
@@ -219,21 +222,24 @@ class K8sModule extends BaseModule {
         throw new ValidationError(`Another worker with the same name ${worker.name} already exists.`);
       workers_names.push(worker.name);
 
-      if (network) break;
       const nodeTwinId = await this.capacity.getNodeTwinId(worker.node_id);
       const features = await this.rmb.request([nodeTwinId], "zos.system.node_features_get", "", 20, 3);
       if (features.some(item => item.includes("zmachine-light") || item.includes("network-light"))) {
-        network = new ZNetworkLight(options.network.name, options.network.ip_range, this.config);
-        await network.load();
+        if (!network) {
+          network = new ZNetworkLight(options.network.name, options.network.ip_range, this.config);
+          await network.load();
+        }
         contractMetadata = JSON.stringify({
           version: 4,
           type: "kubernetes",
           name: options.name,
           projectName: this.config.projectName || `kubernetes/${options.name}`,
         });
       } else {
-        network = new Network(options.network.name, options.network.ip_range, this.config);
-        await network.load();
+        if (!network) {
+          network = new Network(options.network.name, options.network.ip_range, this.config);
+          await network.load();
+        }
```

Comment:
why is this part repeated?

Reply:
I tried to create a function in utils that do those steps but the functon will take the network argument, network, and network type so will be more lines repeated as well 

example
``` ts 

awat createOrLoadNetowrk(options.network.name, options.network.ip_range, this.config, network, 'ZNetworkLight`)

```

Reply:
the case you might have only workers if you add a worker to an existing cluster